packages/riipen-ui/src/components/Backdrop.test.jsx

```diff
@@ -0,0 +1,82 @@
+import { mount } from "enzyme";
+import toJson from "enzyme-to-json";
+import React from "react";
+
+import Backdrop from "./Backdrop";
+
+describe("<Backdrop>", () => {
+  it("renders without errors", () => {
+    let error;
+    try {
+      mount(<Backdrop />);
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toEqual(undefined);
+  });
+
+  describe("default props", () => {
+    it("sets correct default props", () => {
+      const wrapper = mount(<Backdrop />);
+
+      expect(wrapper.find("Backdrop").props().open).toEqual(false);
+      expect(toJson(wrapper)).toMatchSnapshot();
+    });
+  });
+
+  describe("children prop", () => {
+    it("has given child", () => {
+      const child = <h1>The children prop</h1>;
+
+      const wrapper = mount(<Backdrop children={child} />);
+
+      expect(
+        wrapper
+          .find("Backdrop")
+          .children()
+          .contains(child)
+      ).toEqual(true);
+    });
+  });
+
+  describe("classes prop", () => {
+    it("applies classes to the root node of Backdrop", () => {
+      const classes = ["classOne", "classTwo"];
+
+      const wrapper = mount(<Backdrop classes={classes} />);
+
+      expect(
+        wrapper
+          .find("Backdrop")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+      expect(
+        wrapper
+          .find("Backdrop")
+          .childAt(0)
+          .hasClass(classes[1])
+      ).toEqual(true);
+    });
+
+    it("gives an error when classes are provided as one string", () => {
+      const classes = "classOne classTwo";
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<Backdrop classes={classes} />);
+
+      expect(errors).toHaveBeenCalled();
+    });
+  });
+
+  describe("open prop", () => {
+    it("sets open prop to correct value", () => {
+      const open = true;
+
+      const wrapper = mount(<Backdrop open={open} />);
+
+      expect(wrapper.find("Backdrop").props().open).toEqual(true);
+    });
+  });
```

Comment:
I think we could do a better job with this test. The `open` prop actually gets applied to the root node's `className` via the `clsx` package. You might want to read through that package's documetation, but basically if you pass it a `boolean`, it will render the variable's name into the `className`. So in our case, if `open = true`, the `className` of the root node will `hasClass("open")`. Does that make sense?

We should also add the `false` test to make sure when `open = false` that the root node does not get the `open` class applied to it.

Reply:
when I look at the snapshot for `mount(<mount(<Backdrop open={true} />)`, I get this:
```
<withClasses(Backdrop)
  open={true}
>
  <Backdrop
    classes={
      Array [
        "riipen",
        "riipen-backdrop",
      ]
    }
    open={true}
  >
    <div
      aria-hidden={true}
      className="jsx-1867827232 root riipen riipen-backdrop"
    />
    <JSXStyle
      id="1867827232"
    />
  </Backdrop>
</withClasses(Backdrop)>
```
`wrapper.find("Backdrop").hasClass('open')` also gives me a false. Is it possible Backdrop is not applying the open prop properly?

Reply:
Oh woah, this is actually a sort of "bug". Good catch here. Now that I look closer, in the `<style jsx>` element, there actually isn't a `.open` class! This means we probably had it at one point, but remove it.

We should just fully remove the `open` prop as it actually doesn't do anything!

Great catch.